pipeline/pipelines/code_generation_pipeline.py

```diff
@@ -71,4 +73,23 @@ def additional_tasks_for_remote_execution(self, **kwargs):
                                          inject=kwargs)]
 
     def validate_kwargs(self, **kwargs):
-        _validate_codegen_kwargs([], **kwargs)
+        _validate_codegen_kwargs(self.task_factory.get_validate_kwargs(),
+                                 **kwargs)
+
+
+class TaskFactoryBase(object):
+    """ Abstract base class for task factories.
```

Comment:
(whoops, comments are not showing up in-line for some reason. copy-pasting here, sorry for the duplication.)

nit: No space between start of docstring and first line, and no hanging indent on the docstring (see example from google style guide)